### PR TITLE
Cloud 2002.0.21

### PR DESCRIPTION
**Pull request by meker12**
This contains the updates for the [Cloud 2002.0.21 release](https://magento2.atlassian.net/projects/MAGECLOUD/issues/MAGECLOUD-4062).

- [x]  Add changes for MAGECLOUD-4071 ( new CONSUMERS_WAIT_FOR_MAX_MESSAGES env variable)
- [ ]  final dev review
- [x]  whatsnew updated with links
- [ ]  editor review
- [x] YAML check
- [x]  link check

whatsnew
Updated the Cloud Release notes for [ece-tools version 2002.0.21](https://devdocs.magento.com/guides/v2.3/cloud/release-notes/cloud-tools.html#v2002021):

- New environment variables for Cloud deployment: [CONSUMERS_WAIT_FOR_MAX_MESSAGES](https://devdocs.magento.com/guides/v2.3/cloud/env/variables-deploy.html#consumers_wait_for_max_messages) and [LOCK_PROVIDER](https://devdocs.magento.com/guides/v2.3/cloud/env/variables-deploy.html#lock_provider)

- Added a [service compatibility table](https://devdocs.magento.com/guides/v2.3/cloud/project/project-conf-files_services.html#services) that shows support for installed software and service versions such as Elasticsearch,  RabbitMq, Redis, and DB on each Magento version.

- Updated the [Launch Docker](https://devdocs.magento.com/guides/v2.3/cloud/docker/docker-config.html) procedure to remove references to the deprecated `db:docker:convert` command and Docker ENV files which are no longer generated during the Docker build. 

- Added [Docker images](https://devdocs.magento.com/guides/v2.3/cloud/docker/docker-development.html) to support PHP 7.3 and Varnish Cache 6.2.

- Updated the Docker PHP images to add Node.js to support node, npm, and the grunt-cli capabilities inside the [PHP container](https://devdocs.magento.com/guides/v2.3/cloud/docker/docker-development.html#php-fpm-container).

### CODE REVIEW

**Comment by hguthrie** [comment]
don't forget the new lock provider variable in the whatsnew ^^

File: guides/v2.2/cloud/project/project-conf-files_services.md [hunk 1]
```diff
@@ -146,3 +146,18 @@ The current default storage amount per project is 5GB, or 5120MB. You can distri
...
+
+## Service versions
+
+The following table lists the services used in {{site.data.var.ece}} and their version compatibility with the [Magento Cloud template](https://github.com/magento/magento-cloud).
+
+Service   |  Magento 2.3  | Magento 2.2
+--------- | ------------- | ------------
+`elasticsearch` | Magento 2.3.1 and later—1.7, 2.4, 5.2, 6.5<br>Magento 2.3.0—1.7, 2.4, 5.2 | Magento versions 2.2.8 and later—1.7, 2.4, 5.2, 6.5<br>Magento 2.2.0 to 2.2.7—1.7, 2.4, 5.2
+`mariadb` | 10.0 to 10.2  | 10.0 to 10.2
+`nginx`   | 1.9           | 1.9
```
[the remark this answers is not in the document]
Comment (meker12) [reply guides/v2.2/cloud/project/project-conf-files_services.md]: @hguthrie  @YPyltiai  Should we leave nginx version here?

File: test.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[35mTesting Markdown style with mdl ...[0m
```
[the remark this answers is not in the document]
Comment (meker12) [reply test.txt]: Removed

File: guides/v2.2/cloud/project/project-conf-files_services.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (hguthrie) [reply guides/v2.2/cloud/project/project-conf-files_services.md]: This table is not about the ability to change on Cloud, but rather what is compatible with the Cloud environment. I confirmed the chart with Sasha, but if there is some ambiguity here, we can find some way to clarify.

File: guides/v2.2/cloud/release-notes/cloud-tools.md [hunk 3]
```diff
@@ -14,457 +14,501 @@ The following updates describe the latest improvements to the `{{site.data.var.c
...
+
+- {:.fix}<!-- MAGECLOUD-3806 -->Fixed an issue when running the `setup:upgrade` command that did not interrupt the deployment process when a failure occurred during the `app:config:import` task.
+
+- {:.fix}<!-- MAGECLOUD-3866 -->Fixed an issue affecting WARM_UP_PAGES functionality that uses a default store URL. Now, if the `config:show:default-url` command cannot fetch a base URL, then the URL from the MAGENTO_CLOUD_ROUTES variable is used.
+
+- {:.new}<!-- MAGECLOUD-3871 -->Changed the default log level for the file handler to `debug` to reduce the amount of detail in the log displayed in the Project Web Interface, while still providing detailed information for debugging.
+
+- {:.fix}<!-- MAGECLOUD-3913 -->Fixed an issue that can cause database deadlock errors when the `consumers_runner` cron job starts multiple instances of the same consumer on different nodes. Now, if you have enabled the [CRON_CONSUMERS_RUNNER]({{ page.baseurl }}/cloud/env/variables-deploy.html#cron_consumers_runner) deploy variable in your environment, the `consumers_runner` job uses the `single-thread` option to start one instance of each consumer on only one node.
```
[the remark this answers is not in the document]
Comment (meker12) [reply guides/v2.2/cloud/release-notes/cloud-tools.md]: I'll make this update after this branch gets published to avoid new review/approvals.

**Comment by meker12** [comment]
running tests

**Pull request merged**